### PR TITLE
Garrison Plumbing

### PR DESCRIPTION
**Pull request by zach2good**
My confrontation helper from the AMK Cardian fight established some very useful patterns. Garrison is pretty easy 👀 

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

File: scripts/globals/garrison.lua [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+    -- Build npcList
+    local npcList = {}
+    for _, npcId in ipairs(garrisonZoneData.npcs) do
+        table.insert(npcList, npcId)
+    end
+
+    spawnMobs(npcList)
+
+    -- Apply confrontation effect
```
Comment (TeoTwawki) [inline scripts/globals/garrison.lua:76]: Unless you got a cap showing me its confrontation effect (which means SE updated it) it is not confrontation effect :P

I did tons of these before that effect even existed in the client, they def did not use that effect.

**Comment by zach2good** [comment]
I haven't forgotten about this, I just want to get a couple of caps of Valkurm Garrison before I continue

**Comment by RunAwayDSP** [comment]
I have been working on this pr adding a lot of the data to the garrison data... There is a lot of mob data that doesnt have spawn data (getting captures as soon as this weekend). As well as the missing npc data. However, there is still a lot of missing logic structure including the trades to initiate, the commented parts, mannequin drops to all mobs, and treasure for the win. 

If anyone has already captured or added this logic I would love to work with you in getting the rest of this completed.

**Comment by RunAwayDSP** [comment]
Things I have recently done
Added Wave entries for all garrison mobs in garrison_data.lua (many are still missing captures for spawn points)
Added More Traders to Dunes (For testing still need captures for accurate info)
Removed the Table for NPCs instead uses the 1st ID to build off of (npcs = 17201192)

Mobs now spawn with hate to NPCs
NPCs spawn = Number of Players
Each Wave Spawns after previous Wave is killed (Some issues with this as dead but still waiting to depop mobs wont spawn in new wave)
Killing 4th wave = Win (no loot but triggers win condition)

TODO:
all NPC's dead lose condition
Apply Level Sync
Apply Battlefield Effect
Scale Waves/Improve Wave Logic Overall
Wait times as needed
Drop Lists
Treasure Lists
Trade to Start Garrison
Verify No existing Garrison
Prevent Garrison during Ballista
Require trader to be from nation that controls outpost
30 Minute Time Limit
Accurate Capture data for spawn locations, number of spawns, wave order, npc for each zone etc

**Comment by RunAwayDSP** [comment]
Latest Updates
all NPC's dead lose condition
Apply Level Restriction
Apply Battlefield Effect
Drop Lists
Loot Tables (need rates)
Trade to Start Garrison


TODO:
Improve Guard interactions
Add trade trigger to more guards 
Scale Waves/Improve Wave Logic Overall
Wait times as needed
Verify No existing Garrison
Prevent Garrison during Ballista
Require trader to be from nation that controls outpost
30 Minute Time Limit
Accurate Capture data for spawn locations, number of spawns, wave order, npc for each zone etc


Still need more retail captures!!!

File: scripts/globals/garrison.lua (hunk 1 above)
Comment (detetivepro15) [reply scripts/globals/garrison.lua:76]: Eu Sou tudo de bom

**Comment by zach2good** [comment]
This will get superceeded by https://github.com/LandSandBoat/server/pull/1796, which will likely go into `feature/garrison` for a while so multiple people can work on it/test it.

**Pull request closed without merging**